pkg/controller/deployment/deployment_controller.go

```diff
@@ -1294,3 +1413,27 @@ func (dc *DeploymentController) rollbackToTemplate(deployment *extensions.Deploy
 	d, err = dc.updateDeploymentAndClearRollbackTo(deployment)
 	return
 }
+
+// isScalingEvent checks whether the provided deployment has been updated with a scaling event
+// by looking at the desired-replicas annotation in the active replica sets of the deployment.
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
+	if err != nil {
+		return false
+	}
+	allRSs := append(oldRSs, newRS)
+	for _, rs := range controller.FilterActiveReplicaSets(allRSs) {
+		if desiredString, ok := rs.Annotations[deploymentutil.DesiredReplicasAnnotation]; ok {
```

Comment:
> The controller should be able to recognize scaling events.

Why do we need to know a scaling event happened vs just recognize that the aggregate + individual scale values on the RCs are out of sync with what is desired by the deployment?  As I understand it, the replicaCount and podTemplate maybe updated in a single etcd update.  Or a second update to etcd could be made before the sync is triggered for the first update.  It seems we are just normalizing from deployments to replica sets.  I am not sure that this distinction requires any implementation changes, since it appears this is already what we are doing, but it is simpler to describe the system in a way that does not require identifying specific events having occurred.